### PR TITLE
[coroutine] update to 1.5.0

### PR DESCRIPTION
**Pull request by luncliff**
### Changes

New release which allows x86 with better documentation

### Related Issues

* #9624 (`ms-gsl` update by @Farwaykorse )
* #6245 (Previous release)

### Triplet/Compiler
The followings are tested with CI / Local environment

* `x64-windows`: MSVC (vc141, vc142)
* `x86-windows`: MSVC (vc141, vc142)
* `x64-windows-static`: MSVC (vc141, vc142)
* `x86-windows-static`: MSVC (vc141, vc142)
* `x64-osx`: AppleClang 11
* `x64-linux`: Clang 8.0.1+, GCC 10.0.1

### CODE REVIEW

**Comment by luncliff** [comment]
Triplet `WindowsStore` build will be updated within this weekend

File: ports/coroutine/portfile.cmake [hunk 1]
```diff
@@ -1,38 +1,38 @@
 include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/coroutine/portfile.cmake:1]: This line is no longer needed.
Could you please remove it?

File: ports/coroutine/portfile.cmake [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
 
 if(${VCPKG_TARGET_ARCHITECTURE} MATCHES x86)
-    message(FATAL_ERROR "This library doesn't support x86 arch. Please use x64 instead. If it is critical, create an issue at the repo: github.com/luncliff/coroutine")
+    message(WARNING "This library may not work correctly in x86 arch. Please consider using x64 instead")
```
Comment (NancyLi1013) [inline ports/coroutine/portfile.cmake:4]: Could you please help confirm if this version has supported x86 arch?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @luncliff 
Thanks for this PR.
It seems that `coroutine` currently doesn't support uwp and linux.
If so, could you please add `Supports: !(uwp|linux)` to CONTROL file and add `vcpkg_fail_port_install(ON_TARGET "uwp" "linux")` to portfile.cmake?

File: ports/coroutine/portfile.cmake (hunk 2 above)
Comment (luncliff) [reply ports/coroutine/portfile.cmake:4]: The warning is because of code generation with older VS2017 x86/Release build. I will remove that :)

**Comment by luncliff** [comment]
> Hi @luncliff
> Thanks for this PR.
> It seems that coroutine currently doesn't support uwp and linux.
> If so, could you please add Supports: !(uwp|linux) to CONTROL file and add vcpkg_fail_port_install(ON_TARGET "uwp" "linux") to portfile.cmake?

UWP install was fine with my laptop, but seems like I missed something. 
I will take a look at it.

**Comment by NancyLi1013** [comment]
Ok. Thanks for your update.

**Comment by NancyLi1013** [comment]
Could you please also try to fix the regressions on Windows and OSX?

**Comment by luncliff** [comment]
> Could you please also try to fix the regressions on Windows and OSX?

Sorry @NancyLi1013 . I was quite busy. Will update ASAP.

**Comment by NancyLi1013** [comment]
Okay, thanks for your update.

**Comment by NancyLi1013** [comment]
I noticed that it has passed on x86-windows.
Could you please remove `coroutine:x86-windows=fail` from baseline?

**Comment by luncliff** [comment]
@NancyLi1013 Sure. I will update the part after Linux/OSX build reports their ends.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for making a neat library and submitting this update PR!

I relaxed some of the restrictions around Linux; here's a brief idea of how they should be used:
- The `Supports:` field should be optimistic -- if there is a way to use it on Linux (via clang, libc++, etc), it shouldn't be listed as unsupported
- `vcpkg_fail_port_install()` should similarly be optimistic
- `scripts/ci.baseline.txt` must exactly match our specific CI setup; if we aren't using the right GCC version in CI, then it gets marked as failing here

This ensures that users who opt in to jumping through the appropriate hoops (having GCC 10, etc) can access those niche scenarios with vcpkg.

**Review by luncliff — COMMENTED** [review]
learned a lot from this. Thanks :D

**Comment by NancyLi1013** [comment]
@luncliff 
Thanks for your kindly reminder.
I closed it just now.

**Pull request merged**